### PR TITLE
[validator_earning] Monitor for validator earnings

### PR DESCRIPTION
**Pull request by rhazberries**
http://monitor.hmny.io/validators

### CODE REVIEW

File: validator_earning/app/routes.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+html_dis = path.join(data, 'earning.html')
+raw_data = path.join(data, 'validator_info.json')
+net_stat = path.join(data, 'network_stats.json')
+
+@app.route('/validators')
+def validator_info():
+
+    address = request.args.get('address', '')
+    if address != '':
```
Comment (fxfactorial) [inline validator_earning/app/routes.py:17]: https://www.python.org/dev/peps/pep-0572/#id9

File: validator_earning/app/routes.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        if path.exists(raw_data):
+            with open(raw_data, 'r') as f:
```
Comment (fxfactorial) [inline validator_earning/app/routes.py]: i like from daniels' PR the explicit reading as `utf-8`, so lets do the same

File: validator_earning/app/routes.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            return '{"error":"missing data file"}'
+
+    stats = request.args.get('stats', '')
+    if stats == 'true':
+        if path.exists(net_stat):
+            with open(net_stat, 'r') as f:
+                out = ''.join([x.strip() for x in f])
+            return out
+        else:
+            return '{"error":"missing data file"}'
```
Comment (fxfactorial) [inline validator_earning/app/routes.py]: do it with some respargs or something, becuase giving back plain string is just the resp body, doesn't tell the client with a header that is content-type json. yes most clients can just blindly try to parse, see what happens. doesn't matter, purpoe is to be correct in semantic intent. you're giving back json, so set the header resp right

File: validator_earning/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Flask
```
Comment (fxfactorial) [inline validator_earning/requirements.txt:1]: lets pick specific versions maybe, so this doesn't break under us later

File: validator_earning/app/routes.py (hunk 1 above)
Comment (rhazberries) [reply validator_earning/app/routes.py:17]: Using Python 3.7

File: validator_earning/requirements.txt (hunk 4 above)
Comment (rhazberries) [reply validator_earning/requirements.txt:1]: Will fix requirements for all projects in a separate PR.

File: validator_earning/app/routes.py (hunk 3 above)
Comment (rhazberries) [reply validator_earning/app/routes.py]: Added a TODO to enable "POST" requests using jsonrpcserver.
Having `curl` work is fine for now.

**Review by fxfactorial — APPROVED** [review]

**Pull request merged**